src/main/java/bio/terra/service/filedata/azure/tables/TableFileDao.java

```diff
@@ -96,53 +91,11 @@ public FireStoreFile retrieveFileMetadata(TableServiceClient tableServiceClient,
         .collect(Collectors.toList());
   }
 
-  <V> void scanTableObjects(TableClient tableClient, ApiFutureGenerator<V, TableEntity> generator) {
-    ListEntitiesOptions options = new ListEntitiesOptions();
-    PagedIterable<TableEntity> entities = tableClient.listEntities(options, null, null);
-    entities.stream()
-        .forEach(
-            entity -> {
-              try {
-                retryOperation(entity, generator);
-              } catch (InterruptedException e) {
-                throw new FileSystemExecutionException("operation failed", e);
-              }
-            });
-  }
-
-  <T, V> void retryOperation(V input, ApiFutureGenerator<T, V> generator)
-      throws InterruptedException {
-    int retry = 0;
-    while (true) {
-      try {
-        generator.accept(input).get();
-        break;
-      } catch (ExecutionException ex) {
-        final long retryWait =
-            (long) Math.min(SLEEP_BASE_SECONDS * Math.pow(2.5, retry), MAX_SLEEP_SECONDS);
-        retry++;
-        if (retry > AZURE_STORAGE_RETRIES) {
-          throw new FileSystemExecutionException(
-              "Operation failed after " + AZURE_STORAGE_RETRIES + " tries.");
-        } else {
-          logger.warn(
-              "Error in Azure storage table future get - input: {} message: {}",
-              input.toString(),
-              ex.getMessage());
-          logger.info(
-              "Operation will attempt retry #{} after {} millisecond pause.", retry, retryWait);
-        }
-        // Exponential backoff
-        TimeUnit.SECONDS.sleep(retryWait);
-      }
-    }
-  }
-
   void deleteFilesFromDataset(
       TableServiceClient tableServiceClient, InterruptibleConsumer<FireStoreFile> func)
       throws InterruptedException {
     TableClient tableClient = tableServiceClient.getTableClient(TABLE_NAME);
-    scanTableObjects(
+    azureTableUtils.scanTableObjects(
```

Comment:
This API uses futures to perform async operations. The SDK also supports reactive programming as an FYI, and I'm bringing this up just in case the async model is flexible for the project. So you can rewrite this with a "subscription" to the deleted event. This approach simplifies the implementation and removes the need of having to wire up an executor. More information here: [Asynchronous programming in the Azure SDK for Java | Microsoft Docs](https://docs.microsoft.com/en-us/azure/developer/java/sdk/async-programming)

Reply:
That's really cool!

Reply:
For now I added retry options to the TableServiceClient and BlobContainerClient (and removed the `retryOperation` method, but I do want to revisit this model in a future PR